keyboards/redox/proton_c/config.h

```diff
@@ -0,0 +1,84 @@
+// Copyright 2022 xerootg
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x4D44 // "MD"
+#define PRODUCT_ID      0x5244 // "RD"
+#define DEVICE_VER      0x0101 // added 1 for the rev1 + proton c
+#define MANUFACTURER    Falbatech
+#define PRODUCT         The Redox Keyboard
+
+/* key matrix size */
+// Rows are doubled-up
+#define MATRIX_ROWS 10
+#define MATRIX_COLS 7
+
+// wiring of each half
+#define MATRIX_ROW_PINS { B5, B3, B2, B1, B0 }
+#define MATRIX_COL_PINS { A1, A0, B8, B13, B14, B15, B9 }
+// #define MATRIX_COL_PINS { B6, B2, B3, B1, F7, F6, F5} //uncomment this line and comment line above if you need to reverse left-to-right key order
+
+/* COL2ROW or ROW2COL */
+#define DIODE_DIRECTION COL2ROW
+
+/* define if matrix has ghost */
+//#define MATRIX_HAS_GHOST
+
+/* number of backlight levels */
+// #define BACKLIGHT_LEVELS 3
+
+/* Set 0 if debouncing isn't needed */
+#define DEBOUNCE 5
+
+/* serial.c configuration for split keyboard */
+#define SERIAL_USART_FULL_DUPLEX   // Enable full duplex operation mode.
+#define SERIAL_USART_TX_PIN B6     // USART TX pin
+#define SERIAL_USART_RX_PIN B7     // USART RX pin
+// #define USART1_REMAP             // Remap USART TX and RX pins on STM32F103 MCUs, see table below.
+// #define SERIAL_USART_PIN_SWAP    // Swap TX and RX pins if keyboard is master halve.
+                                   // Check if this feature is necessary with your keyboard design and available on the mcu.
+#define SELECT_SOFT_SERIAL_SPEED 1 // or 0, 2, 3, 4, 5
+                                   //  0: 460800 baud
+                                   //  1: 230400 baud (default)
+                                   //  2: 115200 baud
+                                   //  3: 57600 baud
+                                   //  4: 38400 baud
+                                   //  5: 19200 baud
+#define SERIAL_USART_DRIVER SD1    // USART driver of TX and RX pin. default: SD1
+#define SERIAL_USART_TX_PAL_MODE 7 // Pin "alternate function", see the respective datasheet for the appropriate values for your MCU. default: 7
+#define SERIAL_USART_RX_PAL_MODE 7 // Pin "alternate function", see the respective datasheet for the appropriate values for your MCU. default: 7
+#define SERIAL_USART_TIMEOUT 20    // USART driver timeout. default 20
+
+/* Mechanical locking support. Use KC_LCAP, KC_LNUM or KC_LSCR instead in keymap */
+#define LOCKING_SUPPORT_ENABLE
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
+
+/* ws2812 RGB LED */
+#define RGB_DI_PIN A9
```

Comment:
Did you actually test RGB LEDs with that board?

Apparently using the RGB connector on the PCB won't really work, because it would get only 3.3V power, which won't be enough for RGB LEDs. And the slave half gets only VCC from the master half, which would be 3.3V, therefore no chance to get RGB LEDs working there without some PCB hacking (that might be dangerous, because any error may result in getting 5V on the MCU VCC pins and damaging the MCU).

And even if you solve the LED power issues, you may then run into the logic level issues, because the 3.3V logic levels that the MCU outputs may not be compatible with what the LEDs powered from 5V expect.  Actually the Proton-C board has a solution for that (the A3_5V pin), but that's another wire that you would need connect directly to the controller, and another piece of configuration that you need to override (#14490 does exactly that).  Note also that A9 is not 5V tolerant on STM32F303, therefore connecting that pin to 5V-powered RGB LEDs might not be 100% safe even if the logic levels turn out to be acceptable for your particular LEDs.

Also if you actually have working LEDs, you should try adding `WS2812_DRIVER = pwm` and the required configuration for that driver (the default bitbang driver may not work well on a split board, because it disables interrupts for a long time, and that may interfere with the split communications).

Reply:
I don't have any LEDs to test with. I only translated the pin number. I can just remove the definition if its too questionable, but I have seen WS2812's run on 3v3 logic with 5v power.

Reply:
What should I do with this edit? I can't drop it since its defined in rev1's `config.h` with a bad pin assignment for this MCU. I don't have any `ws2812`'s handy to test this with, but considering the number of `esp32`'s driving them, I'm certain they can run just fine on 3v3 logic.

Reply:
Just keep it then; it would probably work, and any RGB LEDs on that board are some DIY work anyway.